src/core/transaction.c

```diff
@@ -393,8 +406,13 @@ static int transaction_verify_order_one(Transaction *tr, Job *j, Job *from, unsi
                         /* Check if this in fact was the beginning of the cycle */
                         if (k == j)
                                 break;
+
+                        prev = k;
                 }
 
+                if (dot && !strextend(&dot, "}"))
+                        log_oom();
```

Comment:
If we fail here, we'll write an invalid output. This should probably be handled the same as above.